src/main.cpp

```diff
@@ -0,0 +1,88 @@
+#include <iostream>
+#include <thread>
+#include <mutex>
+#include <condition_variable>
+#include <vector>
+#include <chrono>
+#include <cmath>
+
+std::mutex mutex;  // Mutex to synchronize access to shared data
+std::condition_variable cv;  // Condition variable to signal consumer
+bool allProducersFinished = false;  // Flag to indicate if all producers finished
+int activeProducers = 0;  // Counter for active producers
+
+/**
+ * @brief Function executed by each producer thread.
+ */
+void producerTask()
+{
+    double value = static_cast<double>(std::hash<std::thread::id>{}(std::this_thread::get_id()));
+    bool lastMessage = false;
+
+    {
+        std::unique_lock<std::mutex> lock(mutex);
+        std::cout << "Producer: " << value << " (Thread ID: " << std::this_thread::get_id() << ") sent." << std::endl;
```

Comment:
The producer needs to actually send data to the consumer. The exercise demand was that the consumer will be the one to print the data